src/llmtuner/model/loader.py

```diff
@@ -93,9 +92,9 @@ def load_model_and_tokenizer(
             model.load_state_dict(vhead_params, strict=False)
             logger.info("Loaded valuehead from checkpoint: {}".format(vhead_path))
 
+    model = model.to(model_args.compute_dtype) if not getattr(model, "quantization_method", None) else model
```

Comment:
We cannot cast the adapter weights to half-precision, move this line before initializing the adapters in training mode.